File: CalendarWithObjC/CalendarWithObjC/CustomCalendar/CustomCalendarCell/CalendarCollectionViewCell.m
//
//  CalendarCollectionViewCell.m
//  CathayBank
//
//  Created by Marvin iOS on 2018/9/13.
//  Copyright © 2018年 YuChen Hsu. All rights reserved.
//

#import "CalendarCollectionViewCell.h"

@implementation CalendarCollectionViewCell

- (void)setSelectedState {
    _dateLabel.hidden = NO;
    _dateLabel.textColor = UIColor.whiteColor;
    _indicateView.hidden = NO;
    CGFloat width = self.frame.size.width;
    _indicateView.layer.cornerRadius = width / 2.0;
    self.statusType = CalendarCollectionViewCellTypeForSelected;
}

- (void)setNormalState {
    _dateLabel.hidden = NO;
    _dateLabel.textColor = UIColor.blackColor;
    _indicateView.hidden = YES;
    self.statusType = CalendarCollectionViewCellTypeForNormal;
}

- (void)setEmptyUI {
    [_dateLabel setHidden:YES];
    _dateLabel.hidden = YES;
    _indicateView.hidden = YES;
}

- (void)setUnableState {
    _dateLabel.hidden = NO;
    _dateLabel.textColor = UIColor.lightGrayColor;
    _indicateView.hidden = YES;
    self.statusType = CalendarCollectionViewCellTypeForUnable;
}

@end

File: CalendarWithSwift/CalendarWithSwift/CustomCalendar/CustomCalendarCell/CalendarCollectionViewCell.swift
//
//  CalendarCollectionViewCell.swift
//  CalendarWithSwift
//
//  Created by YuChen Hsu on 2019/7/22.
//  Copyright © 2019 YuChen Hsu. All rights reserved.
//

import UIKit

enum CalendarCollectionViewCellType: Int {
    case calendarCollectionViewCellTypeForUnable = 1
    case calendarCollectionViewCellTypeForNormal = 2
    case calendarCollectionViewCellTypeForSelected = 3
}

class CalendarCollectionViewCell: UICollectionViewCell {
    var statusType : CalendarCollectionViewCellType = .calendarCollectionViewCellTypeForUnable
    @IBOutlet weak var dateLabel: UILabel!
    @IBOutlet weak var indicateView: UIView!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }
    
    func setSelectedState() {
        dateLabel.isHidden = false
        dateLabel.textColor = UIColor.white
        indicateView.isHidden = false
        let width : CGFloat = self.frame.size.width
        indicateView.layer.cornerRadius = width / 2.0
        statusType = .calendarCollectionViewCellTypeForSelected
    }

    func setNormalState() {
        dateLabel.isHidden = false
        dateLabel.textColor = UIColor.black
        indicateView.isHidden = true
        statusType = .calendarCollectionViewCellTypeForNormal
    }
    
    func setEmptyUI() {
        dateLabel.isHidden = true
        indicateView.isHidden = true
    }
    
    func setUnableState() {
        dateLabel.isHidden = false
        dateLabel.textColor = UIColor.lightGray
        indicateView.isHidden = true
        statusType = .calendarCollectionViewCellTypeForUnable
    }
}
